Sprint-2/5-stretch-extend/format-time.js

```diff
@@ -3,11 +3,18 @@
 // Your task is to write tests for as many different groups of input data or edge cases as you can, and fix any bugs you find.
 
 function formatAs12HourClock(time) {
-  const hours = Number(time.slice(0, 2));
-  if (hours > 12) {
-    return `${hours - 12}:00 pm`;
+  const [hourStr, minuteStr] = time.split(":");
+  const hours = Number(hourStr);
+
+  let period = hours >= 12 ? "pm" : "am";
+  let displayHour = hours % 12;
+  if (displayHour === 0) {
+    displayHour = 12; // Handles 00:xx (midnight) and 12:xx (noon) 
   }
-  return `${time} am`;
+  // If you input "08:00" is allowed to pass
+  const paddedHour = hourStr.length === 2 && displayHour < 10 ? `0${displayHour}` : `${displayHour}`;
```

Comment:
Nice work in formatting the hour it is returned. I have a resource that you might want to consider should you need to do this again in the future

[String Padding](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart)